storybookjs/storybook
# Build: Generate version file with preval macro

Issue: #11220

## What I did

I split out the preval macro code into its own commit from #11220. Couple open questions:
1. Why does the e2e PnP tests fail?
2. Does compilation use the babel cache? If so, then https://github.com/kentcdodds/babel-plugin-preval/issues/19 says we can't use this method.

## How to test

- Is this testable with Jest or Chromatic screenshots? e2e tests seem to catch it
- Does this need a new example in the kitchen sink apps? ❌ 
- Does this need an update to the documentation? ❌ 

If your answer is yes to any of these, please make sure to include it in your PR.
